### PR TITLE
RCIAM-245_propagate_vopersonverified_to_SP

### PR DESCRIPTION
**Pull request by ioigoume**
- Refactored getProfile query.
- Construct,add voPersonVerifiedEmail into state.

### CODE REVIEW

File: lib/Auth/Process/COmanageDbClient.php [hunk 1]
```diff
@@ -438,6 +444,59 @@ private function getProfile($personId)
...
+        if (!empty($attributes['given'])) {
+          $state['Attributes']['givenName'] = array($attributes['given']);
+        }
+        if (!empty($attributes['family'])) {
+          $state['Attributes']['sn'] = array($attributes['family']);
+        }
+        if (!empty($attributes['mail'])) {
+          // XXX for the mail attribute always get the first
+          $state['Attributes']['mail'] = array(
+            explode(',', $attributes['mail'])[0]
```
Comment (NicolasLiampotis) [inline lib/Auth/Process/COmanageDbClient.php:472]: How does the sql query determine the order of the returned mail attributes ?

**Review by NicolasLiampotis — CHANGES_REQUESTED** [review]
1. We can minimise the risk of returning multi-valued given/family and email attributes by limiting results to "official" 
2. We need to make sure that attributes are returned in the same order
3. Fix style of new code according to https://github.com/simplesamlphp/simplesamlphp/blob/master/CONTRIBUTING.md, i.e. use 4 spaces (not tabs)

File: lib/Auth/Process/COmanageDbClient.php [hunk 2]
```diff
@@ -438,6 +444,59 @@ private function getProfile($personId)
...
+
+        }
+        if (!empty($attributes['edupersonscopedaffiliation'])) {
+          $state['Attributes']['eduPersonScopedAffiliation'] = explode(',', $attributes['edupersonscopedaffiliation']);
+        }
+        if (!empty($attributes['identifier'])) {
+          $state['Attributes']['uid'] = array($attributes['identifier']);
+        }
+      }
+      return true;
```
[the remark this answers is not in the document]
Comment (ioigoume) [reply lib/Auth/Process/COmanageDbClient.php:497]: Return false. The profile could be empty. In that case we return false. So for consistency, if everything goes smoothly we return true.

File: lib/Auth/Process/COmanageDbClient.php [hunk 3]
```diff
@@ -438,6 +444,59 @@ private function getProfile($personId)
...
+   * Get and construct the profile of the COPerson.
+   * The profile includes the attributes:
+   * given, sn, mail, voPersonVerifiedEmail, eduPersonScopedAffiliation, uid
+   *
+   * @param array &$state
+   * @param integer $co_person_id
+   * @return boolean
+   * @throws Exception
+   */
+  private function constructProfile(&$state, $co_person_id) {
```
[the remark this answers is not in the document]
Comment (ioigoume) [reply lib/Auth/Process/COmanageDbClient.php:457]: Done.

[the remark this answers is not in the document]
Comment (ioigoume) [reply lib/Auth/Process/COmanageDbClient.php:457]: The identifier and name attributes will never be multivalued due to COmanage. The name marked as primary can only be one pair of given, sn attributes. Also we can not specify a second uid attribute with the same value. Added a comment in the internal ticket as well.

File: lib/Auth/Process/COmanageDbClient.php (hunk 1 above)
Comment (ioigoume) [reply lib/Auth/Process/COmanageDbClient.php]: The mail attribute will always contain the mail with the smallest row id value in the database.

Comment (NicolasLiampotis) [reply lib/Auth/Process/COmanageDbClient.php]: Effectively that means the oldest mail attribute value, right?

**Review by NicolasLiampotis — CHANGES_REQUESTED** [review]
Fix code style of `constructProfile` according to https://github.com/simplesamlphp/simplesamlphp/blob/master/CONTRIBUTING.md, i.e. use 4 spaces

**Review by NicolasLiampotis — APPROVED** [review]
LGTM

**Pull request merged**